src/realtime_metrics/main.py

```diff
@@ -6,17 +6,168 @@
+
+
+def eta_accuracy(stop_time_updates: list[tuple[TripUpdate, StopTimeUpdate]]):
+    # computes the accuracy of the given stop time updates using the eta bucketing approach
+
+    # initialize buckets with zero correct and incorrect samples each
+    buckets: list[list] = []
+    buckets.append([0,0])
+    buckets.append([0,0])
+    buckets.append([0,0])
+    buckets.append([0,0])
+    
+    # sort stop time updates into the correct buckets
+    for update in stop_time_updates:
+        trip_update = update[0]
+        stop_time_update = update[1]
+        actual_arrival_time = get_actual_arrival_time(trip_update=trip_update, stop_time_update=stop_time_update)
+        time_variance = actual_arrival_time - trip_update.timestamp.timestamp()
+
+        bucket_index = 0
+        upper_limit = 0
+        lower_limit = 0
+        if time_variance <= 0:
+            # remove stop time updates published after arrival
+            continue
+        elif time_variance <= 180:
+            # 0-3 min bucket
+            bucket_index = 0
+            upper_limit = 90
+            lower_limit = -30
+        elif time_variance <= 360:
+            # 3-6 min bucket
+            bucket_index = 1
+            upper_limit = 150
+            lower_limit = -60
+        elif time_variance <= 600:
+            # 6-10 min bucket
+            bucket_index = 2
+            upper_limit = 210
+            lower_limit = -60
+        elif time_variance <= 900:
+            # 10-15 min bucket
+            bucket_index = 3
+            upper_limit = 270
+            lower_limit = -90
+        else: 
+            # too far in the past
+            #print("Stop Time Update is outside the allowed interval.")
+            continue
+
+        # check if inside allowed interval
+        actual_delay = get_actual_delay(trip_update=trip_update, stop_time_update=stop_time_update)
+        predicted_delay = stop_time_update.arrival_delay
+        # TODO: fix calculation
+        difference =  predicted_delay - actual_delay
+
+        bucket = buckets[bucket_index]
+
+        if difference <= upper_limit and difference >= lower_limit:
+            bucket[0] += 1
+        bucket[1] += 1
+        
+        buckets[bucket_index] = bucket
+
+    logger.debug("buckets: %s", buckets)
+
+    # compute accuracy in each bucket
+    accuracies = []
+    for i in range(0,len(buckets)):
+        bucket = buckets[i]
+        correct = bucket[0]
+        total = bucket[1]
+        if total == 0:
+            continue
+        accuracies.append(correct / total)
+
+    # compute total accuracy
+    if len(accuracies) == 0:
+        logger.info("No data provided!")
+        return
+    mean_accuracy = numpy.mean(accuracies) * 100
+    logger.info(f"ETA accuracy: {mean_accuracy}%")
+
+
+def get_actual_delay(trip_update: TripUpdate, stop_time_update: StopTimeUpdate) -> int:
+    # return the actual delay for the route and stop of the given TripUpdate
+    #return -numpy.random.randint(-90, 270)
```

Comment:
```suggestion

"""
returns the actual delay for the route and stop of the given TripUpdate
"""
def get_actual_delay(trip_update: TripUpdate, stop_time_update: StopTimeUpdate) -> int:
```
Use docstring instead of the comment here